ANMN/burst_averaged_product/burst_average_product.py

```diff
@@ -0,0 +1,388 @@
+                setattr(output_netcdf_obj['%s_burst_sd' % var], var_att, getattr(input_netcdf_obj[var], var_att))
+
+        var_units = get_imos_parameter_info(var, 'units')
+        if var_units:
+            setattr(output_netcdf_obj[var], 'units', var_units)
+            setattr(output_netcdf_obj['%s_burst_min' % var], 'units', var_units)
+            setattr(output_netcdf_obj['%s_burst_max' % var], 'units', var_units)
+            setattr(output_netcdf_obj['%s_burst_sd' % var], 'units', var_units)
+
+        if get_imos_parameter_info(var, 'standard_name'):
+            setattr(output_netcdf_obj['%s_num_obs' % var], 'standard_name', "%s number_of_observations" % get_imos_parameter_info(var, 'standard_name'))
+
+        setattr(output_netcdf_obj['%s_num_obs' % var], 'units', "1")
+        if var != 'DEPTH':
+            setattr(output_netcdf_obj[var], "coordinates" , "TIME LATITUDE LONGITUDE DEPTH")
+
+        # set up var values
+        output_netcdf_obj[var][:]                  = np.ma.masked_invalid(burst_vars[var][1])
+        output_netcdf_obj['%s_burst_min' % var][:] = np.ma.masked_invalid(burst_vars[var][2])
+        output_netcdf_obj['%s_burst_max' % var][:] = np.ma.masked_invalid(burst_vars[var][3])
+        output_netcdf_obj['%s_burst_sd' % var][:]  = np.ma.masked_invalid(burst_vars[var][4])
+        output_netcdf_obj['%s_num_obs' % var][:]   = np.ma.masked_invalid(burst_vars[var][5])
+
+    # add gatts and variable attributes as stored in config files
+    conf_file_generic = os.path.join(os.path.dirname(__file__), 'generate_nc_file_att')
+    generate_netcdf_att(output_netcdf_obj, conf_file_generic, conf_file_point_of_truth = True)
+
+    if instrument_type == 'WQM':
+        conf_file = os.path.join(os.path.dirname(__file__), 'generate_wqm_nc_file_gatt')
+    elif instrument_type == 'CTD':
+        conf_file = os.path.join(os.path.dirname(__file__), 'generate_ctd_nc_file_gatt')
+    generate_netcdf_att(output_netcdf_obj, conf_file)
+
+    # append original comment gatt to burst average comment gatt
+    if hasattr(input_netcdf_obj, 'comment'):
+        gatt = 'comment'
+        setattr(output_netcdf_obj, gatt, '%s. %s' % (getattr(input_netcdf_obj, gatt), getattr(output_netcdf_obj, gatt) ))
+
+    # add values to variables
+    output_netcdf_obj['TIME'][:]      = np.ma.masked_invalid(time_burst_vals)
+    output_netcdf_obj['LATITUDE'][:]  = np.ma.masked_invalid(input_netcdf_obj['LATITUDE'][:])
+    output_netcdf_obj['LONGITUDE'][:] = np.ma.masked_invalid(input_netcdf_obj['LONGITUDE'][:])
+
+    output_netcdf_obj.close()
+    input_netcdf_obj.close()
+
+    shutil.move(output_netcdf_file_path, output_dir)
+    shutil.rmtree(tmp_netcdf_dir)
+    return os.path.join(output_dir, os.path.basename(output_netcdf_file_path))
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        msg = 'Usage: %s INPUT_FILE OUTPUT_DIR\n' % sys.argv[0]
+        raise RuntimeError(msg)
```

Comment:
Actually, you don't really need to raise an error here. Just print the message to stderr and exit.